### DISCUSSION TITLE
How to implement with existing data / legacy applications?

### DISCUSSION THREAD

**Discussion by DanjBethel** [spatie/laravel-event-sourcing#323] [Q&A] [answered]
Was wondering if anyone could shed some light on how I would implement event sourcing with existing data. Allow me to elaborate.

I currently have a client project that focuses on scheduling employees to different locations. They have existing scheduling data that I have imported into the application. The application itself is finished, but I would like to now refactor to use event sourcing in order to keep track of employees scheduling hours which will be used in payroll calculation. I have event sourcing working for new events, but how would I generate stored events for the data that already exist? I thought about perhaps creating a command that fires an event for each database entry that I wish to capture, but how would I do that without actually recreating the entry itself?

Am I maybe looking at this all wrong? Any advice would be greatly appreciated.

P.S: New to event sourcing. First timer! Be gentle.

**Comment by freekmurze** [comment, marked as answer]
We've created an entire course on this subject. You might find it helpful: https://event-sourcing-laravel.com

**Reply by DanjBethel** [reply]
Purchased. Thank you

**Reply by DanjBethel** [reply]
Is it possible for you to point me in the right direction in the ebook that answers my original question of implementing event sourcing with existing data? Or is the video course different? Not sure if the content of the ebook vs the video course is the same or not.

Thank you. Loving the ebook so far!
